src/nvim/strings.c

```diff
@@ -496,6 +496,20 @@ char *vim_strchr(const char *const string, const int c)
   }
 }
 
+// Sized version of strchr that can handle embedded NULs.
+// Adjusts n to the new size.
+char *strnchr(const char *p, size_t *n, int c)
+{
+  while (*n > 0) {
+    if (*p == c) {
+      return (char *)p;
+    }
+    p++;
+    (*n)--;
+  }
+  return NULL;
+}
```

Comment:
Is this really needed? This seems mostly the same of `memchr()`, except that it takes a pointer to `n`.

Reply:
If you can replace with `memchr`, then go for it. I gave it a generic name, because it is a generic function.